fontbe/src/orchestration.rs

```diff
@@ -377,6 +401,48 @@ impl Persistable for Kerning {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq)]
+pub enum PairPosEntry {
+    Pair(GlyphId, ValueRecordBuilder, GlyphId, ValueRecordBuilder),
+    Class(GlyphSet, ValueRecordBuilder, GlyphSet, ValueRecordBuilder),
+}
+
+impl PairPosEntry {
+    pub(crate) fn add_to(self, builder: &mut PairPosBuilder) {
+        match self {
+            PairPosEntry::Pair(gid0, rec0, gid1, rec1) => {
+                builder.insert_pair(gid0, rec0, gid1, rec1)
+            }
+            PairPosEntry::Class(set0, rec0, set1, rec1) => {
+                builder.insert_classes(set0, rec0, set1, rec1)
+            }
+        }
+    }
+}
+
+/// A chunk of kerning that needs to be fed into a [PairPosBuilder]
+#[derive(Default, Clone, Serialize, Deserialize, PartialEq)]
+pub struct KernSegment {
```

Comment:
for glyf/gvar (and also for the KerningSegment's timing label) you used the word "fragment" instead of "segment", might make sense to be consistent